Add IO for writing to local file system

javajawa opened the pull request:
The IOConfig only supplies an Output, which can create and delete files, including writing or appending content.

javajawa commented:
I found some notes:
- Add test for creating dirs
- Add test for binary write
- Add test for attempting to write to `../` or `/abs` style paths
- Add test for non-existent output dir
- Add test for non-initialised output dir

ajCameron commented on src/mewbot/io/file_storage.py:
```diff
@@ -0,0 +1,162 @@
+class WriteToFileOutputEvent(FSOutputEvent):
+    """
+    Writes the given contents to the specified file.
+
```
Suggest adding something like "within the designated _output_path" - just to emphasize that arbitary file creation is not supported.

ajCameron commented on src/mewbot/io/file_storage.py:
```diff
@@ -0,0 +1,162 @@
+class DeleteFileOutputEvent(FSOutputEvent):
+    """
+    Deletes a file with the given name.
+    """
```
Again suggest adding "inside the designated _output_path"

ajCameron commented:
All looks good apart from some very minor doc suggestions.

exachixkitsune commented:
Does this pull remain valid, and/or want to wait until #120 is done to comply with that system?

javajawa commented:
> Does this pull remain valid, and/or want to wait until #120 is done to comply with that system?

Merging will wait for whatever change actually implements the plugin system (that's where my series of merges are slowly approaching base off of #127), but moving it to use the plugin system is pretty simple.

ajCameron commented:
What the...

ajCameron commented:
Okay. Name clash between a remote branch and a local branch.

Annoying. But fixable.

javajawa committed:
Add IO for writing to local file system

The IOConfig only supplies an Output, which can create and delete files,
including writing or appending content.

javajawa committed:
Add requirements-dev to install_deps

javajawa committed:
Move pytest config to pyproject.toml

javajawa committed:
Add missing copyright header

javajawa commented:
Handled in https://github.com/mewbotorg/mewbot-io-file-storage/pull/3
